crates/ink/macro/src/storage/storable.rs

```diff
@@ -108,6 +125,32 @@ fn storable_enum_derive(s: &synstructure::Structure) -> TokenStream2 {
              }
          }
     });
+
+    let encoded_size_body = s.variants().iter().enumerate().map(|(index, variant)| {
+        let pat = variant.pat();
+        let index = index as u8;
+        let fields = variant.bindings().iter().map(|field| {
+            let span = field.ast().ty.span();
+            quote_spanned!(span =>
+                encoded_size = encoded_size
+                    .checked_add(::ink::storage::traits::Storable::encoded_size(#field))
+                    .unwrap();
+            )
+        });
+        quote! {
+             #pat => {
+                {
+                    encoded_size = encoded_size
+                        .checked_add(<::core::primitive::u8 as ::ink::storage::traits::Storable>::encoded_size(&#index))
```

Comment:
Could you use `expect` instead of `unwrap`, please? At least, it allows us to know the place of the error.

Also, maybe it is better to use `saturating_add` instead.

Reply:
Hmm, good catch! `saturating_add` won't panic, but unwrapping here after `checked_add` defeats the whole point. I think `saturating_add` is not really what we want, as the decoding would not fit in case of saturation. I think I'll change the `encoded_size` method to return an `Option<usize>` instead, reflecting the situation. Thanks!

Reply:
Decided to go with `saturating_add`; in reality the buffer size can't be larger than the the pointer size anyways.